app/models/page.rb

```diff
@@ -20,15 +20,23 @@ def self.are_there_any_hidden_pages?
     Page.exists?( hidden: true )
   end
 
-  # TODO: Return the default top-level page
+  # Return the default top-level page
   def self.default_page
-    # 1.  Page.find( Config.default_page )
-    # 2.  Section.find( Config.default_section )
-    # 2a. default_section.pages.default_page
-    # 2b. default_section.pages.first
+    page = find_default_page
+    return page if page
+
+    section = PageSection.default_section
+    page    = section&.pages&.default_page
+    page  ||= section&.pages&.first
+    return page if page
 
     return Page.top_level_pages.first if Page.top_level_pages.first
 
     Page.first
   end
+
+  def self.find_default_page
+    name = Setting.get 'default_page'
+    Page.where( name: name ).or( Page.where( slug: name ) ).first
```

Comment:
Consider raising an error if there is more than one matching `Page`, it seems like an invalid state that would allow inconsistent behaviour.

Reply:
Should `find_default_page` exclude hidden pages?